packages/api/lambdas/sqs-message-remover.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const get = require('lodash.get');
+
+const {
+  deleteSQSMessage,
+  sqs
+} = require('@cumulus/common/aws');
+const {
+  getSfEventMessageObject,
+  getSfEventStatus,
+  isFailedSfStatus,
+  isSfExecutionEvent,
+  isTerminalSfStatus
+} = require('@cumulus/common/cloudwatch-event');
+const log = require('@cumulus/common/log');
+
+
+/**
+ * Determine if the SQS queue update is needed for the event
+ *
+ * Return false if:
+ *   - Event has no workflow status
+ *   - Workflow is not in a terminal state
+ *   - Event message has no 'meta.eventSource' property or eventSource type is not 'sqs'
+ *   - Event message property meta.eventSource.deleteCompletedMessage is not true
+ *   - Event message property meta.eventSource.workflow_name is not the same as meta.workflow_name
+ *
+ * @param {Object} event - A workflow execution event
+ * @returns {boolean} True if SQS queue update is needed
+ */
+function isSqsQueueUpdateNeeded(event) {
+  const eventStatus = getSfEventStatus(event);
+  const eventMessage = getSfEventMessageObject(event, 'input', '{}');
+
+  if (!isSfExecutionEvent(event)
+    || !isTerminalSfStatus(eventStatus)
+    || get(eventMessage, 'meta.eventSource.type') !== 'sqs'
+    || get(eventMessage, 'meta.eventSource.deleteCompletedMessage', false) !== true
+    || get(eventMessage, 'meta.eventSource.workflow_name') === null
+    || get(eventMessage, 'meta.eventSource.workflow_name') !== get(eventMessage, 'meta.workflow_name')) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Update SQS queue when workflow of the message is completed
+ *
+ * @param {Object} event - Cloudwatch event
+ */
+async function updateSqsQueue(event) {
+  if (!isSqsQueueUpdateNeeded(event)) return 'Not a valid event for updating SQS queue';
```

Comment:
This is returning a string, but we expect this function to return an object. I would either log  that message and return `Promise.resolve()` or throw an error.

Reply:
I updated the jsdoc.  It should return a `Promise` object.